crates/relayer-cosmos/src/base/impls/transaction.rs

```diff
@@ -218,7 +218,8 @@ fn split_events_by_messages(in_events: Vec<AbciEvent>) -> Vec<Vec<Arc<AbciEvent>
 
     for event in in_events.into_iter() {
         if event.kind == "message"
-            && event.attributes.len() == 1
+            // TODO: What is the purpose of this filter ?
+            //&& event.attributes.len() == 1
```

Comment:
Why this change?

Reply:
I added more details to the reason behind the change. Let me know if it is clear enough or if I should add more

Reply:
All good, thanks!